hazelcast/src/main/java/com/hazelcast/internal/partition/operation/MigrationRequestOperation.java

```diff
@@ -260,6 +202,58 @@ protected void readInternal(ObjectDataInput in) throws IOException {
         fragmentedMigrationEnabled = in.readBoolean();
     }
 
+    private final class OffloadImpl extends Offload {
+        private OffloadImpl() {
+            super(MigrationRequestOperation.this);
+        }
+
+        @Override
+        public void start() {
+            verifyMasterOnMigrationSource();
+
+            Address source = migrationInfo.getSource();
+            Address destination = migrationInfo.getDestination();
+            verifyExistingTarget(nodeEngine, destination);
+
+            if (destination.equals(source)) {
+                getLogger().warning("Source and destination addresses are the same! => " + toString());
+                setFailed();
```

Comment:
Currently `setFailed()` doesn't fail with an exception but expects `success = false` response to be sent automatically after operation returns. But with `Offloaded`, response won't be sent automatically after `start()` is called.

Reply:
If you leave `MigrationRequestOperation` out of scope for now, I can do a cleanup after this PR is merged.

Reply:
Sure. No problem. I'll revert it.